model/chunks/chunkassignment.go

```diff
@@ -28,25 +47,29 @@ func (a *Assignment) Verifiers(chunk *flow.Chunk) flow.IdentifierList {
 // HasVerifier checks if a chunk is assigned to the given verifier
 // TODO: method should probably error if chunk has unknown index
 func (a *Assignment) HasVerifier(chunk *flow.Chunk, identifier flow.Identifier) bool {
-	assignedVerifiers, found := a.verifiersForChunk[chunk.Index]
-	if !found {
+	if chunk.Index >= uint64(len(a.verifiersForChunk)) {
 		// is verifier assigned to this chunk?
 		// No, because we only assign verifiers to existing chunks
 		return false
 	}
+	assignedVerifiers := a.verifiersForChunk[chunk.Index]
 	_, isAssigned := assignedVerifiers[identifier]
 	return isAssigned
 }
 
-// Add records the list of verifier nodes as the assigned verifiers of the chunk
-// it returns an error if the list of verifiers is empty or contains duplicate ids
-func (a *Assignment) Add(chunk *flow.Chunk, verifiers flow.IdentifierList) {
+// Add records the list of verifier nodes as the assigned verifiers of the chunk.
+// Requires chunks to be added in order of their Index (increasing by 1 each time).
+// It returns an error if the list of verifiers is empty or contains duplicate ids
```

Comment:
There are existing tests that either pass in duplicate IDs (TestAssignment/TestAssignDuplicate) or an empty list (TestVerificationHappyPath) - unclear to me if we actually want this to be an error and update those tests, or not

Reply:
Thanks for pointing this out. The current documentation is inaccurate, since no error is returned. 

I don't think we need to explicitly check for duplicate verifier IDs. We don't expect them, but they will be removed by the conversion to a map/set anyway. 

Regarding empty lists, I think we should allow this (as the current behaviour does. Running the network with sealing effectively disabled is a valid configuration for testing etc. Chunk-alpha (how many verifiers are assigned to each chunk) [is configurable](https://github.com/onflow/flow-go/blob/85fd812a7e49c921277160d8d31b2eba0a436a57/cmd/consensus/main.go#L158), and zero is a valid configuration value. 

```suggestion
// Panics if chunks are not added in ascending Index order.
```